### PR TITLE
update CircleCI config to 2.0 syntax

### PR DESCRIPTION
**Pull request by jacobmarble**
Expect flakiness described in #9554.

Updating CircleCI config to 2.0 is required by aug-2018, and it gives us access to some fancy CI features.

### CODE REVIEW

**Comment by jacobmarble** [comment]
The first couple of executions with this new config show a small performance improvement (~2m) just from this change. Might be that 1.0 was running in a docker container, while I have the 2.0 config running on a VM, which has 2 cores. Docker Layer Caching would improve performance more dramatically, perhaps by ~8m.

**Comment by jacobmarble** [comment]
@e-dard I should note that this PR removes a lot of logic, for example there is no more `circle-test.sh`, the CircleCI config just runs `test.sh` and doesn't persist STDOUT. I think this is appropriate, please confirm.

**Comment by jacobmarble** [comment]
Latest changes improve (cached) test runtime to ~8m.

File: services/subscriber/service_test.go [hunk 1]
```diff
@@ -185,7 +185,6 @@ func TestService_ModeALL(t *testing.T) {
 }
 
 func TestService_ModeANY(t *testing.T) {
-	t.Skip("TODO: flaky test.")
```
Comment (e-dard) [inline services/subscriber/service_test.go]: Is this implication here that you think moving to this new infrastructure will fix this flaky test?

File: .gitignore [hunk 2]
```diff
@@ -66,6 +66,7 @@ config.toml
 
 # test data files
 integration/migration_data/
+test-logs/
```
Comment (mark-rushakoff) [inline .gitignore]: Is this necessary?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -277,4 +277,4 @@ func BenchmarkSomething(b *testing.B) {
 
 Continuous Integration testing
 -----
-InfluxDB uses CircleCI for continuous integration testing. To see how the code is built and tested, check out [this file](https://github.com/influxdata/influxdb/blob/master/circle-test.sh). It closely follows the build and test process outlined above. You can see the exact version of Go InfluxDB uses for testing by consulting that file.
+InfluxDB uses CircleCI for continuous integration testing. To see how the code is built and tested, check out [this file](https://github.com/influxdata/influxdb/blob/master/test.sh). It closely follows the build and test process outlined above. You can see the exact version of Go InfluxDB uses for testing by consulting that file.
```
Comment (jsternberg) [inline CONTRIBUTING.md]: Should we include instructions for how to use the shell script here? I think this section needs to be updated with the recommended testing procedures that are useful to a contributor. I'm not sure I would recommend using any of our CI scripts for testing in all honesty.

Comment (mark-rushakoff) [inline CONTRIBUTING.md]: The last sentence of this line is false and should be removed.

File: .gitignore (hunk 2 above)
Comment (jacobmarble) [reply .gitignore]: `./test.sh 0` writes to that directory

File: circle.yml [hunk 4]
```diff
@@ -1,23 +0,0 @@
-machine:
-    services:
-        - docker
-    environment:
-      GODIST: "go1.9.2.linux-amd64.tar.gz"
```
[the remark this answers is not in the document]
Comment (jacobmarble) [reply circle.yml]: This has no effect. `./test.sh n` creates a container, installs go (versioned, see `test.sh`) then executes tests.

[the remark this answers is not in the document]
Comment (jacobmarble) [reply circle.yml]: In other words, this installs go on the host container/VM, then executes the tests in guest containers.

**Comment by jacobmarble** [comment]
@mark-rushakoff PTAL

**Comment by jacobmarble** [comment]
FYI a failing version of this PR, just added `t.Fail()` to one test:
https://github.com/influxdata/influxdb/pull/9717

**Pull request merged**